### PR TITLE
Cleanup installs happening in the github workflows

### PR DESCRIPTION
**Pull request by bsweger**
Addresses #404 

The scope of these commits is making sure that all required dependencies are in the package's DESCRIPTION instead of being explicitly installed as a workflow step (for example, DT is required to build the vignettes).

Additionally, there's some cleanup to remove workflow-based installs if they aren't being used and/or reducing the footprint (e.g., explicitly use `ggplot2` in the vignette rather than pull in the entire tidyverse).

Longer term, we have a goal to make these workflows better align to the newer ones used in the hubverse (#406 )

**note:** there are several smaller commits so it's easier to revert any breaking changes...happy to squash after approval to tidy up the commit history

### CODE REVIEW

**Comment by nickreich** [comment]
@bsweger I'm assuming you will ping me when this is ready for a full review, and am holding off on a review until then.

File: .github/workflows/pkgdown.yaml
Comment (bsweger) [inline .github/workflows/pkgdown.yaml:25]: I assumed the `scoringutils` version here should match the changes made in https://github.com/reichlab/covidHubUtils/pull/400

File: vignettes/covidHubUtils-overview.Rmd
Comment (bsweger) [inline vignettes/covidHubUtils-overview.Rmd:75]: Another place where we can be more specific about the required library to avoid a big tidyverse install.

File: vignettes/covidHubUtils-overview.Rmd
Comment (bsweger) [inline vignettes/covidHubUtils-overview.Rmd:436]: `scoringutils` is already installed as part of `covidHubUtils`

**Comment by bsweger** [comment]
@nickreich ready for review!

File: .github/workflows/pkgdown.yaml
Comment (elray1) [inline .github/workflows/pkgdown.yaml:25]:
```suggestion
          devtools::install_github("epiforecasts/scoringutils@v1.2.2")
```

File: DESCRIPTION
Comment (elray1) [inline DESCRIPTION:85]: It looks like tidyr was already listed above, on new line 58?
```suggestion
    RSQLite
```

**Review by elray1 — CHANGES_REQUESTED** [review]
Just a couple of minor changes in comments.

A bigger point is that although the github action definition updates looked plausible to me, I don't know of a way to test that they work without actually merging the changes in and giving them a run... do you?

File: .github/workflows/pkgdown.yaml
Comment (bsweger) [reply .github/workflows/pkgdown.yaml]: Great catch, thank you! 😅

File: DESCRIPTION
Comment (bsweger) [reply DESCRIPTION:85]: ah--good call!

await my forthcoming linting and styling PR in which we shall alphabetize the list of imports

**Comment by bsweger** [comment]
> A bigger point is that although the github action definition updates looked plausible to me, I don't know of a way to test that they work without actually merging the changes in and giving them a run... do you?

@elray1 `r_cmd_check.yaml` and `pr_unittest.yaml` run when a PR opens, so I made sure they worked before opening the PR for review (however, the tests started failing after I committed your suggested changes, so I'll look at that).

is `pkgdown.yaml` something we can run on a PR and then again on merge to main?

I recall that github actions have some capacity for ad-hoc runs, but I haven't looked into it yet.

**Review by nickreich — APPROVED** [review]
It feels like I'm breaking some GitHub code of conduct to give an approving review to a PR that is not passing a check, but I did get a few minutes to look at this and it the changes look good to me. So once the checks are passing I don't have other concerns.

**Comment by bsweger** [comment]
@elray1 I believe I found and fixed the issue that was causing the tests to fail (that said, I'm not sure why they were passing before merging in your requested changes (which seem unrelated..and I reverted them to just see if the tests would start passing again, and they did! :tableflip:)

Anyway, everything is green now, and I squashed the commits so the history is a bit cleaner. Can you re-review when you have a chance? Thanks!

**Review by elray1 — APPROVED** [review]

**Pull request merged**
